lib/pacemaker/pcmk_sched_group.c

```diff
@@ -439,29 +458,31 @@ colocate_with_group(pcmk_resource_t *dependent, const pcmk_resource_t *primary,
             member = (pcmk_resource_t *) primary->priv->children->data;
         }
         if (member == NULL) {
-            return; // Nothing to colocate with
+            return 0;   // Nothing to colocate with
         }
 
-        member->priv->cmds->apply_coloc_score(dependent, member, colocation,
-                                              false);
-        return;
+        return member->priv->cmds->apply_coloc_score(dependent, member,
+                                                     colocation, false);
     }
 
     if (colocation->score >= PCMK_SCORE_INFINITY) {
         pcmk__config_err("%s: Cannot perform mandatory colocation with"
                          " non-colocated group %s",
                          dependent->id, primary->id);
-        return;
+        return 0;
     }
 
     // Colocate dependent with each member individually
     for (const GList *iter = primary->priv->children;
          iter != NULL; iter = iter->next) {
 
         member = iter->data;
-        member->priv->cmds->apply_coloc_score(dependent, member, colocation,
-                                              false);
+        priority_delta += member->priv->cmds->apply_coloc_score(dependent,
+                                                                member,
+                                                                colocation,
+                                                                false);
     }
+    return priority_delta;
```

Comment:
ditto

Reply:
We're the primary in this case. I think anything that gets added to the dependent's priority should happen in the dependent's method, or we run the risk of double-adding...

Originally I wasn't taking a priority delta at all in `colocate_with_group()`, but then I found myself with a 0 priority delta in `colocate_group_with()` while processing a "group with group" colocation. In other words, I'm doing this for the benefit of a possible dependent group further up the call chain.

These colocation call chains are confusing, but that's old news.